### PR TITLE
get the result of handler in packetMiddleware

### PR DESCRIPTION
**Pull request by mhdizade**
Fixes eggjs/egg#3174

### CODE REVIEW

File: lib/io.js [hunk 1]
```diff
@@ -102,14 +102,14 @@ module.exports = app => {
               const ctx = args.splice(-1)[0];
               ctx.args = ctx.req.args = args;
               handler.call(ctx)
-                .then(() => ctx[CtxEventSymbol].emit('finshed'))
-                .catch(e => {
-                  if (e instanceof Error) {
-                    e.message = '[egg-socket.io] controller execute error: ' + e.message;
+                .then((result) => ctx[CtxEventSymbol].emit('finshed', {result}))
```
Comment (thonatos) [inline lib/io.js]:
```js
handler.call(ctx)
  then(result => ctx[CtxEventSymbol].emit('finshed', { result }))
```

File: lib/packetMiddlewareInit.js [hunk 2]
```diff
@@ -24,10 +24,10 @@ module.exports = (app, socket, packet, next, packetMiddlewares, nsp) => {
     if (routerMap && routerMap.has(eventName)) {
       debug('[egg-socket.io] wait controller finshed!');
       // after controller execute finished, resume middlewares
-      await new Promise((resolve, reject) => {
-        ctx[CtxEventSymbol].on('finshed', error => {
+      return new Promise((resolve, reject) => {
+        ctx[CtxEventSymbol].on('finshed', ({result, error}) => {
```
Comment (thonatos) [inline lib/packetMiddlewareInit.js]:
```js
ctx[CtxEventSymbol].on('finshed', ({ result, error }) => {}
```

**Comment by thonatos** [comment]
- https://travis-ci.org/eggjs/egg-socket.io/jobs/451839884

**Comment by ngot** [comment]
https://github.com/eggjs/egg/issues/3174#issuecomment-436966546

**Pull request closed without merging**